python/dgl/convert.py

```diff
@@ -17,397 +18,211 @@
-    formats : str or list of str
-        It can be ``'coo'``/``'csr'``/``'csc'`` or a sublist of them,
-        Force the storage formats.  Default: ``['coo', 'csr', 'csc']``.
-    idtype : int32, int64, optional
-        Integer ID type. Valid options are int32 or int64. If None, try infer from
-        the given data.
-    device : Device context, optional
-        Device on which the graph is created. Default: infer from data.
-    card : int, optional
-        Deprecated (see :attr:`num_nodes`). Cardinality (number of nodes in the graph).
-        If None, infer from input data, i.e. the largest node ID plus 1. (Default: None)
+        The number of nodes in the graph. If not given, this will be the largest node ID
+        plus 1 from the :attr:`data` argument. If given and the value is no greater than
+        the largest node ID from the :attr:`data` argument, DGL will raise an error.
+    idtype : int32 or int64, optional
+        The data type for storing the structure-related graph information such as node and
+        edge IDs. It should be a framework-specific data type object (e.g., ``torch.int32``).
+        If ``None`` (default), DGL infers the ID type from the :attr:`data` argument.
+        See "Notes" for more details.
+    device : device context, optional
+        The device of the returned graph, which should be a framework-specific device object
+        (e.g., ``torch.device``). If ``None`` (default), DGL uses the device of the tensors of
+        the :attr:`data` argument. If :attr:`data` is not a tuple of node-tensors, the
+        returned graph is on CPU.  If the specified :attr:`device` differs from that of the
+        provided tensors, it casts the given tensors to the specified device first.
 
     Returns
     -------
-    DGLHeteroGraph
+    DGLGraph
+        The created graph.
 
-    Examples
-    --------
-    Create from pairs of edges with form (src, dst)
+    Notes
+    -----
+    1. If the :attr:`idtype` argument is not given then:
+
+       - in the case of the tuple of node-tensor format, DGL uses
+         the data type of the tensors for storing node/edge IDs.
+       - in the case of the tuple of sequence format, DGL uses int64.
 
-    >>> g = dgl.graph([(0, 2), (0, 3), (1, 2)])
+       Once the graph has been created, you can change the data type by using
+       :func:`dgl.DGLGraph.long` or :func:`dgl.DGLGraph.int`.
 
-    Create from source and destination vertex ID lists
+       If the specified :attr:`idtype` argument differs from the data type of the provided
+       tensors, it casts the given tensors to the specified data type first.
+    2. The most efficient construction approach is to provide a tuple of node tensors without
+       specifying :attr:`idtype` and :attr:`device`. This is because the returned graph shares
+       the storage with the input node-tensors in this case.
+    3. DGL internally maintains multiple copies of the graph structure in different sparse
```

Comment:
Can we give a ref (external link to dgl user guide) to `graph structure of different sparse formats`?

Reply:
Done.